src/porepy/numerics/fv/mpfa.py

```diff
@@ -449,7 +455,7 @@ def _local_discr(self, g, k, bnd, eta=None, inverter="numba", apertures=None):
             # Rotate the grid into the xy plane and delete third dimension. First
             # make a copy to avoid alterations to the input grid
             g = g.copy()
-            cell_centers, face_normals, face_centers, R, _, nodes = pp.cg.map_grid(g)
+            cell_centers, face_normals, face_centers, R, _, nodes = pp.cg.map_grid(g, tol)
```

Comment:
Why do we suddenly rotate g? I thought the philosophy was to do all computations in true 3d. Isn't that right, @keileg?

Reply:
2D discretizations must be rotated; this has always been the case. This is very deep in the implementation, where there is a tacit assumption that the number of components in the local gradients is equal to the dimension. We can change it, but for now, rotations are needed. I guess the difference now is that @alessiofumagalli has come upon some really nasty grids.